src/core/propertyTypes.js

```diff
@@ -0,0 +1,61 @@
+var coordinates = require('../utils/coordinates');
+
+var propertyTypes = module.exports.propertyTypes = {};
+
+// Built-in property types.
+registerPropertyType('boolean', false, boolParse);
+registerPropertyType('int', 0, intParse);
+registerPropertyType('number', 0, numberParse);
+registerPropertyType('selector', '', selectorParse, selectorStringify);
+registerPropertyType('string', '', defaultParse, defaultStringify);
+registerPropertyType('vec3', { x: 0, y: 0, z: 0 }, coordinates.parse, coordinates.stringify);
+
+/**
+ * Register a parser for re-use such that when someone uses `type` in the schema,
+ * `schema.process` will set the property `parse` and `stringify`.
+ *
+ * @param {string} type - Type name.
+ * @param [defaultValue=null] -
+ *   Default value to use if component does not define default value.
+ * @param {function} [parse=defaultParse] - Parse string function.
+ * @param {function} [stringify=defaultStringify] - Stringify to DOM function.
+ */
+function registerPropertyType (type, defaultValue, parse, stringify) {
+  propertyTypes[type] = {
```

Comment:
We should prevent registering the same type multiple times. We have to check if the type is already registered and if it is throw an error message like we do with component registration: 
https://github.com/aframevr/aframe-core/blob/master/src/core/component.js#L190